plugins/history_plugin/include/eosio/history_plugin/history_plugin.hpp

```diff
@@ -18,6 +18,25 @@ namespace eosio {
 
 namespace history_apis {
 
+class read_write {
+   history_const_ptr history;
+
+   public:
+      read_write(history_const_ptr&& history)
+        : history(history) {}
```

Comment:
I realize you are only following example, but this should use `std::forward` or just take by value and use `std::move`.